__tests__/db.test.js

```diff
@@ -221,3 +221,111 @@ describe("/api/articles/:article_id/comments", () => {
       })
   })
 })
+
+describe("POST /api/articles/:article_id/comments", () => {
+  test("201: Posting a comment returns how it looks in the table in the database", () => {
+    const newComment = {
+      username: "butter_bridge",
+      body: "This is truly a mind-blowing experience",
+    }
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send(newComment)
+      .expect(201)
+      .then(({ body }) => {
+        expect(body.comment.comment_id).toBe(19)
+        expect(body.comment.body).toBe(
+          "This is truly a mind-blowing experience"
+        )
+        expect(body.comment.votes).toEqual(expect.any(Number))
+        expect(body.comment.author).toBe("butter_bridge")
+        expect(body.comment.article_id).toBe(2)
+        expect(body.comment.created_at).toEqual(expect.any(String))
+      })
+  })
+  test("400: Returns an error if a valid id is entered but there is no article", () => {
+    const newComment = {
+      username: "butter_bridge",
+      body: "This is truly a mind-blowing experience",
+    }
+    return request(app)
+      .post("/api/articles/7002/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad request")
+      })
+  })
+  test("400: Returns an error if an invalid id type is entered", () => {
+    const newComment = {
+      username: "butter_bridge",
+      body: "This is truly a mind-blowing experience",
+    }
+    return request(app)
+      .post("/api/articles/threehundredandtwo/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad request")
+      })
+  })
+  test("400: Returns an error input object is missing values", () => {
+    const newComment = {
+      body: "This is truly a mind-blowing experience",
+    }
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad request")
+      })
+  })
+  test("400: Returns an error input object values are the wrong data types", () => {
+    const newComment = {
+      name: "butter_bridge",
+      body: 8,
+    }
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad request")
+      })
+  })
```

Comment:
Need a test for Status 404: Not found, username does not exist
Responds with error object including status and message.